book/forms.md

```diff
@@ -425,14 +433,24 @@ focused):
 class Browser:
     def handle_key(self, e):
         # ...
-        if self.chrome.focus:
-            self.chrome.keypress(e.char)
+        if self.chrome.keypress(e.char):
             self.draw()
         elif self.focus == "content":
             self.active_tab.keypress(e.char)
             self.draw()
 ```
 
+Here I've changed `keypress` to return true if the browser chrome consumed the
+key:
+
+``` {.python}
+class Chrome:
+    def keypress(self, char):
+        if self.focus == "address bar":
+            self.address_bar += char
+        return self.focus == "address bar"
```

Comment:
I know it's a line longer but I think this would be cleaner if we just did `return True` inside the `if` statement and `return False` outside it.